From b4a810c7bd06752c9728675067f63b98b9186ddb Mon Sep 17 00:00:00 2001
From: Mythri P K <mythri.p.k@intel.com>
Date: Fri, 28 Nov 2014 09:21:39 +0530
Subject: [PATCH 01/15] ASoC: Intel: Enabler patch for stitched fw binary

Issue: GMINL-6292

This is an enabler patch for stitched fw binary and
would need corresponding fw changes enabled.

Change-Id: Iaa107a0316062232190b6674f1683e4da3d4aab4
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
Signed-off-by: Fabien Marotte <fabien.marotte@intel.com>
---
 sound/soc/intel/sst/sst.h      | 2 +-
 sound/soc/intel/sst/sst_acpi.c | 8 ++++++--
 sound/soc/intel/sst/sst_dsp.c  | 2 +-
 3 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/sound/soc/intel/sst/sst.h b/sound/soc/intel/sst/sst.h
index d97c549..3a2f092 100644
--- a/sound/soc/intel/sst/sst.h
+++ b/sound/soc/intel/sst/sst.h
@@ -115,7 +115,7 @@ enum sst_stream_states {
 enum sst_ram_type {
 	SST_IRAM	= 1,
 	SST_DRAM	= 2,
-	SST_DDR	= 5,
+	SST_DDR		= 5,
 	SST_CUSTOM_INFO	= 7,	/* consists of FW binary information */
 };
 
diff --git a/sound/soc/intel/sst/sst_acpi.c b/sound/soc/intel/sst/sst_acpi.c
index 0f23ef17..f2fe209a 100644
--- a/sound/soc/intel/sst/sst_acpi.c
+++ b/sound/soc/intel/sst/sst_acpi.c
@@ -163,7 +163,7 @@ static const struct sst_info byt_fwparse_info = {
 
 
 static const struct sst_info cht_fwparse_info = {
-	.use_elf	= true,
+	.use_elf	= false,
 	.max_streams	= MAX_NUM_STREAMS_MRFLD,
 	.dma_max_len	= SST_MAX_DMA_LEN_MRFLD,
 	.iram_start	= SST_BYT_IRAM_PHY_START,
@@ -203,6 +203,10 @@ static const struct sst_lib_dnld_info  byt_lib_dnld_info = {
 #endif
 };
 
+static const struct sst_lib_dnld_info  cht_lib_dnld_info = {
+	.mod_ddr_dnld = false,
+};
+
 static const struct sst_ipc_info cht_ipc_info = {
 	.use_32bit_ops = false,
 	.ipc_offset = 0,
@@ -215,7 +219,7 @@ struct sst_platform_info cht_platform_data = {
 	.bdata = NULL,
 	.pdata = NULL,
 	.ipc_info = &cht_ipc_info,
-	.lib_info = NULL,
+	.lib_info = &cht_lib_dnld_info,
 	.start_recovery_timer = false,
 };
 
diff --git a/sound/soc/intel/sst/sst_dsp.c b/sound/soc/intel/sst/sst_dsp.c
index d6cb08f..d716504 100644
--- a/sound/soc/intel/sst/sst_dsp.c
+++ b/sound/soc/intel/sst/sst_dsp.c
@@ -1501,7 +1501,7 @@ void sst_post_download_mrfld(struct intel_sst_drv *ctx)
 	/* For mrfld, download all libraries the first time fw is
 	 * downloaded */
 	pr_debug("%s: lib_dwnld = %u\n", __func__, ctx->lib_dwnld_reqd);
-	if (ctx->lib_dwnld_reqd) {
+	if (ctx->lib_dwnld_reqd && ctx->info.use_elf != false) {
 		sst_load_all_modules_elf(ctx, sst_modules_mrfld, ARRAY_SIZE(sst_modules_mrfld));
 		ctx->lib_dwnld_reqd = false;
 	}
-- 
1.9.1

